class.jetpack-ixr-client.php

```diff
@@ -32,6 +32,9 @@ function __construct( $args = array(), $path = false, $port = 80, $timeout = 15
 	function query() {
 		$args = func_get_args();
 		$method = array_shift( $args );
+		if ( 'jetpack.deregister' === $method ) {
+			$args['requesting_user_id'] = get_current_user_id();
```

Comment:
It feels like it doesn't make sense to implement such a specific logic in such a generic class.

Seems like we already have a way to specify the user ID - by default it's `0`, and is passed over as `$this-> jetpack_args`. Could we use this method instead? We'd just need to specify it as the first argument when we're instantiating the `Jetpack_IXR_Client` object before sending the `jetpack.deregister` query.

Reply:
Instead of changing the IXR class at all, can we instead explicitly add the user ID to the deregister call itself?

```diff
diff --git a/class.jetpack.php b/class.jetpack.php
index 27dc7ca18..04f5623c0 100644
--- a/class.jetpack.php
+++ b/class.jetpack.php
@@ -3288 +3288 @@ p {
-			$xml->query( 'jetpack.deregister' );
+			$xml->query( 'jetpack.deregister', get_current_user_id() );
```